taskiq/compat.py

```diff
@@ -0,0 +1,33 @@
+# flake8: noqa
+from typing import Any, TypeVar
+
+import pydantic
+
+pydantic_version = tuple([int(part) for part in pydantic.__version__.split(".")])
```

Comment:
Actually it's a potential bug.

Your solution will work for normal versions, but not for pre-releases and other non-ordinary versions. Also, there's a small (almost zero) possibility that pydantic will rename or move `__version__`.

So, here's my suggestion. Consider using importlib_metadata and packaging.version. Here's a simple example:

```python
In [1]: from packaging.version import Version

In [2]: Version("2.0.1-pre") >= Version("2.0")
Out[2]: True

In [3]: Version("1.22.34.599999") >= Version("2.0")
Out[3]: False
```

As you see, it doesn't fail for alpha versions. 

So, you'll do something like this:

```python
from importlib_metadata import version
from packaging.version import Version, parse

PYDANTIC_VER = parse(version("pydantic"))

if PYDANTIC_VER >= Version("2.0"):
    ...
```

Also, note. DO NOT USE `importlib.metadata`, use [importlib_metadata](https://pypi.org/project/importlib-metadata/) package for compatibility with older versions of python.

Reply:
nice suggestion, thank you 👍